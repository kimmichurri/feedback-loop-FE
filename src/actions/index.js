export const hasError = (message) => ({
  type: 'HAS_ERROR',
  message
})

export const isLoading = (boolean) => ({
  type: 'IS_LOADING',
  isLoading: boolean
})

export const setSurvey = (survey) => ({
  type: 'SET_SURVEY',
  survey
})

export const setCohorts = (cohorts) => ({
  type: 'SET_COHORTS',
  cohorts
})

export const setUser = (user) => ({
  type: 'SET_USER',
  user
})

export const setCurrentCohort = (cohort) => ({
  type: 'SET_CURRENT_COHORT',
  cohort
})