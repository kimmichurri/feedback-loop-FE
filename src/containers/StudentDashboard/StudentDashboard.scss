.dashboard-container {
  
}

.student-survey-container {
  display: flex;
  justify-content: space-between;
}

.student-surveys,
.student-feedback {
  display: flex;
  flex-direction: column;
  align-items: center;
  width: 45%;
  height: 60vh;
}

.article-container {
  display: flex;
  flex-direction: column;
  width: 90%;
  overflow: scroll;
}

.student-dashboard-header {
  font-size: 34px;
  height: 51px;
  width: 100%;
  margin: 0px;
  color: $main-grey;
  line-height: 135%;
  text-align: left;
  font-weight: normal;
}

.article-header {
  width: 100%;
  margin: 10px 0px;
  border-bottom: solid $accent-red 2px;
  color: $main-grey;
  font-size: 22px;
  font-weight: normal;
  padding: 10px 0px;
}

.student-pending-survey {
  background-color: $accent-red;
  border-radius: 5px;
  border: none;
  color: $white;
  font-family: $main-font;
  font-weight: bold;
  font-size: 16px;
  height: 50px;
  width: 100%;
  margin: 10px 0px;
  &:hover {
    cursor: pointer;
    background-color: $accent-red-hover;
  }
  &:focus {
    outline: none;
  }
}

