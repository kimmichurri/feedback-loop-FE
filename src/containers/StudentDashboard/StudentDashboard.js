import React, { Component } from 'react'
import { connect } from 'react-redux'
import { handleGet } from '../../thunks/handleGet'
import StudentSurvey from '../StudentSurvey/StudentSurvey'
import { setStudentSurveys } from '../../actions/'

export class StudentDashboard extends Component {
  

  async componentDidMount() {
    const url = `https://turing-feedback-api.herokuapp.com/api/v1/surveys/pending?api_key=${this.props.user}`
    const surveys = await this.props.handleGet(url)
    this.props.setStudentSurveys(surveys)
  }

  renderSurvey = (survey) => {
    this.props.history.push(`/student-survey/${survey.surveyName}`)
    return <StudentSurvey key={survey.id}/>
  }
 
  render() {
    let surveyButtons
    if (this.props.studentSurveys) {
      surveyButtons = this.props.studentSurveys.map(survey => {
        return <button key={survey.id} onClick={() => {this.renderSurvey(survey)}}>{survey.surveyName}</button>
      })
    }
    
    return(
      <div className='dashboard-container'>
        Student Dashboard
        {surveyButtons}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user,
  studentSurveys: state.studentSurveys
})

export const mapDispatchToProps = (dispatch) => ({
  handleGet: (url) => dispatch(handleGet(url)),
  setStudentSurveys: (surveys) => dispatch(setStudentSurveys(surveys))
})

export default connect(mapStateToProps, mapDispatchToProps)(StudentDashboard)
