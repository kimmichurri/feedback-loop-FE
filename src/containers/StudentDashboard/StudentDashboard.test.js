import React from 'react'
import { shallow } from 'enzyme'
import { StudentDashboard, mapStateToProps, mapDispatchToProps } from './StudentDashboard'
jest.mock('../../thunks/handleGet')
import { setStudentSurveys } from '../../actions'

describe('StudentDashboard', () => {
  let wrapper
  let mockUser
  let mockStudentSurveys
  let mockHandleGet
  let mockSetStudentSurveys

  beforeEach(() => {
    mockUser = '12345'
    mockStudentSurveys = [
      { id: 1, name: "Survey 1"},
      { id: 2, name: "Survey 2"}
    ]
    mockHandleGet = jest.fn()
    mockSetStudentSurveys = jest.fn()
    wrapper = shallow(
      <StudentDashboard
        user={mockUser}
        studentSurveys={mockStudentSurveys}
        handleGet={mockHandleGet}
        setStudentSurveys={mockSetStudentSurveys}
      />
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('should return the expected state as props', () => {
      const mockState = {
        user: '12345',
        studentSurveys: [
          { id: 1, name: "Survey 1"},
          { id: 2, name: "Survey 2"}
        ],
        fakeState: 'fakeState'
      }
      const expected = {
        user: '12345',
        studentSurveys: [
          { id: 1, name: "Survey 1"},
          { id: 2, name: "Survey 2"}
        ]
      }
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should return handleGet to dispatch', () => {
      const mockUrl = 'www.getsurveys.com'
      const mockDispatch = jest.fn()
      const handleGet = jest.fn()
      const actionToDispatch = handleGet(mockUrl)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.handleGet(mockUrl)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

    it('should return setStudentSurveys to dispatch', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = setStudentSurveys(mockStudentSurveys)
      const mappedProps = mapDispatchToProps(mockDispatch)

      mappedProps.setStudentSurveys(mockStudentSurveys)

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })

  })

})