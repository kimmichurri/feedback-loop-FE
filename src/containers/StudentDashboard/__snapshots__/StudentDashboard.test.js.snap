// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StudentDashboard should match the snapshot 1`] = `
<div
  className="dashboard-container"
>
  <button
    className="response-button"
    key="1"
    onClick={[Function]}
  />
  <button
    className="response-button"
    key="2"
    onClick={[Function]}
  />
</div>
`;
